extensions/contract-agreement-transfer-api/src/main/java/de/sovity/edc/extension/contractagreementtransferapi/controller/ContractAgreementTransferRequestDto.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright (c) 2023 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.contractagreementtransferapi.controller;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import jakarta.validation.constraints.NotNull;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+
+@JsonDeserialize(builder = ContractAgreementTransferRequestDto.Builder.class)
+public class ContractAgreementTransferRequestDto {
+
+    private String id;
+    @NotNull(message = "contractId cannot be null")
+    private String contractAgreementId;
```

Comment:
or rather, this belongs in the url

The API Model should only contain information that is relevant, and exactly the relevant information. Either we use only DataAddress or we create an object that contains the difference, as in, only the fields that won't be overwritten by us.

As in, violates single object responsibility.